src/main/scala/mesosphere/marathon/core/task/termination/impl/KillServiceActor.scala

```diff
@@ -66,8 +66,11 @@ private[impl] class KillServiceActor(
   }
 
   override def preStart(): Unit = {
-    context.system.eventStream.subscribe(self, classOf[InstanceChanged])
+    // TODO: this actor needs to initialized once with all instances with goals Stopped/Decommissioned.
```

Comment:
Should this be done in this PR?

Reply:
🤷‍♀️ @meichstedt ?

Reply:
I think so, yes. Otherwise the actor won't retry after a failover.